File: src/main/kotlin/com/github/sam0delkin/intellijpsa/model/CompletionModel.kt
package com.github.sam0delkin.intellijpsa.model

import com.fasterxml.jackson.annotation.JsonProperty
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class CompletionModel {
    val text: String? = null
    val bold: Boolean? = null

    @SerialName("presentable_text")
    @JsonProperty("presentable_text")
    val presentableText: String? = null

    @SerialName("tail_text")
    @JsonProperty("tail_text")
    val tailText: String? = null
    val type: String? = null
    val priority: Double? = null
    val link: String? = null
}

@Serializable
class NotificationModel {
    var type: String = ""
    var text: String = ""
}

@Serializable
class CompletionsModel {
    var completions: List<CompletionModel>? = null
    val notifications: List<NotificationModel>? = null
}

File: phpStubs/psa-server-sdk/src/CompletionModel.php
<?php

namespace Psa;

final class CompletionModel
{
    /** @var string|null */
    public $text;

    /** @var bool */
    public $bold = false;

    /** @var string|null */
    public $presentableText;

    /** @var string|null */
    public $tailText;

    /** @var string|null */
    public $type;

    /** @var int|null */
    public $priority;

    /** @var string|null */
    public $link;

    public function toArray(): array
    {
        $data = ['bold' => $this->bold];

        if ($this->text !== null) {
            $data['text'] = $this->text;
        }
        if ($this->presentableText !== null) {
            $data['presentable_text'] = $this->presentableText;
        }
        if ($this->tailText !== null) {
            $data['tail_text'] = $this->tailText;
        }
        if ($this->type !== null) {
            $data['type'] = $this->type;
        }
        if ($this->priority !== null) {
            $data['priority'] = $this->priority;
        }
        if ($this->link !== null) {
            $data['link'] = $this->link;
        }

        return $data;
    }
}
